tests/functional/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+#include <olp/authentication/Settings.h>
+#include <olp/authentication/TokenProvider.h>
+#include <olp/core/cache/DefaultCache.h>
+#include <olp/core/client/CancellationToken.h>
+#include <olp/core/client/Condition.h>
+#include <olp/core/client/HRN.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/core/logging/Log.h>
+#include <olp/core/porting/make_unique.h>
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <testutils/CustomParameters.hpp>
+
+#include "olp/dataservice/read/PartitionsRequest.h"
+#include "olp/dataservice/read/VolatileLayerClient.h"
+
+using namespace olp::dataservice::read;
+using namespace testing;
+
+namespace {
+
+using namespace olp::dataservice::read;
+const auto kAppIdEnvName = "dataservice_read_test_appid";
+const auto kAppSecretEnvName = "dataservice_read_test_secret";
+const auto kCatalogEnvName = "dataservice_read_test_catalog";
+const auto kLayerEnvName = "dataservice_read_volatile_layer";
```

Comment:
dataservice_read_volatile_layer must be present in scripts/linux/fv/olp-cpp-sdk-functional-test.variables